Jint/Runtime/ExceptionHelper.cs

```diff
@@ -61,7 +61,7 @@ public static T ThrowTypeError<T>(Engine engine, string message = null, Exceptio
             return default;
         }
 
-        public static void ThrowTypeError(Engine engine, string message = null, Exception exception = null)
+        public static Exception ThrowTypeError(Engine engine, string message = null, Exception exception = null)
```

Comment:
should not be needed

Reply:
I've changed this back.  Without it you can't unfortunately use pattern matching eg:

if(thisRef is PromiseInstance promise)
    ExceptionHelper.ThrowTypeError(Engine, "message");

if(promise.Task.Result != null)...

As the compiler (incorrectly) believes that promise can be unassigned.

I've also changed the related methods that call this with "throw ExceptionHelper.ThrowTypeError" to get around this compiler issue.

Reply:
That is why there's generic version for the most ExceptionHelper methods, you can do for example:

`return ExceptionHelper.ThrowTypeError<PromiseInstance>(Engine, $"undefined is not iterable (cannot read property {GlobalSymbolRegistry.Iterator._value})");`

Reply:
Ah of course, so obvious!  Thanks :)